### ISSUE TITLE
Take better timing averages

### ISSUE THREAD

**Issue by john-butler-iv** [john-butler-iv/project-euler-rust#3]
### Background / Problem
Right now, when timing problems, we just take a simple arithmetic mean when outputting the average run time. This can be the right choice, but it makes a lot more sense when your distribution is symmetric. When timing results, though, there is a hard limit towards lower times, and the drop off is pretty steep in most cases. Conversely, there are a bunch of factors that could make a program run slower, and there's not a hard limit to how long it can run, although it's pretty rare to see something terribly far from the mean. Therefore, when timing problems, you will see the raw geometric average skew to a larger number. Here's a graph to illustrate my point. The curvy blue line roughly aligns with the distribution you'd see for run times*, the solid black line represents the most common single run time, and the dashed green line represents what you would get if you took an arithmetic mean.

![image](https://github.com/user-attachments/assets/d75a8d37-c9c5-4ae3-8cd2-66e9d575fd39)

As a caveat, I suppose it's not necessarily true that you _would_ want the peak of the distribution instead of the arithmetic mean, and if the two are related in some matter, then it's not really important which one we use as long as we're consistent when comparing problems.

*I'm not claiming that the run times actually fall under that distribution, I just chose some equation that has roughly the same shape that I would expect.

### Request

Look into ways to create a more reliable estimate of the mode. If that is the number we want to use. Right off the bat, a simple solution could be to take the arithmetic mean of the numbers as we do now, and then filter off the outliers. Maybe only accept values within a certain percentage of the initially calculated average? In the specific distribution I used above, if you ignore values above ~1.87 times greater than the simple arithmetic mean, you should get the peak of the distribution, although you would have to do some analysis on actual run times to find a good value.

This approach assumes that all problems have the same runtime distribution, ignoring scaling, and does not take into account how it's usually slower to run a problem for the first couple iterations due to CPU instruction prefetch and needing to map extra memory for the program and stuff like that.

So to sum up,

1. exclude the first couple runs of a program
2. create a histogram of the run times of real problems, and figure out where to draw the line for outliers
4. validate the approach works for every problem
